userland/test/test_publisher.cpp

```diff
@@ -0,0 +1,48 @@
+#include <chrono>
+#include <iostream>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/Uint32.h>
+
+
+template<typename T>
+int publish(
+  rclcpp::Node::SharedPtr node,
+  void (*set_data_func)(typename T::Ptr&, size_t),
+  size_t number_of_messages
+) {
```

Comment:
We should discuss this particular point of code style again outside of the scope of this pr, because I would have written this as:

```c++
template<typename T>
int publish(
  rclcpp::Node::SharedPtr node,
  void (*set_data_func)(typename T::Ptr&, size_t),
  size_t number_of_messages)
{
  ...
}
```

Such that the opening `{` is visually lined up with its counter part on line 34, rather than the one on line 27, which it is lined up with but does not match. Normally I would say what you wrote above is good because it minimizes the number of lines of delta to add a parameter to the end or parameters, but since you cannot leave a trailing `,` it's not the case here, so it doesn't really make it better to put the closing `)` on it's own line.